lib/ethers/multicall.ex

```diff
@@ -0,0 +1,302 @@
+defmodule Ethers.Multicall do
+  @moduledoc """
+  High-level module providing convenient utilities and an easy-to-use API for interecting with
+  `Multicall3` (more info https://www.multicall3.com).
+
+  This module aggregates multiple **read-only** Ethereum contract calls into a single eth_call
+  which can be passed to `Ethers.call/2`. The response can then be passed to
+  `Ethers.Multicall.decode/2` for decoding responses returned by the `Multicall3` contract.
+
+  ## Examples
+  ```elixir
+  calls = [
+    ContractA.foo(),
+    {ContractB.foo(), to: "0x..."},
+  ]
+
+  calls
+  |> Multicall.aggregate3() # Or can use `Multicall.aggregate2`
+  |> Ethers.call!()
+  |> Multicall.decode(calls)
```

Comment:
Sorry, forgot to add the Ethers prefix in the last comment.

```suggestion
  calls
  |> Ethers.Multicall.aggregate3() # Or can use `Ethers.Multicall.aggregate2/1`
  |> Ethers.call!()
  |> Ethers.Multicall.decode(calls)
```